File: samples/SelfAspNetVb/SelfAspNetApi/Controllers/BooksController.vb
Imports System.Net
Imports System.Web.Http

Namespace Controllers
    Public Class BooksController
        Inherits ApiController
        Function GetBook(ByVal id As String) As Book
            '<HttpGet>
            'Function RetrieveBook(ByVal id As String) As Book
            Dim db = New SelfAsp()
            Dim book As Book = db.Book.Find(id)
            If IsNothing(book) Then
                Throw New HttpResponseException(HttpStatusCode.NotFound)
            End If
            Return book
        End Function

    End Class
End Namespace

File: samples/SelfAspNet/SelfAspNetApi/Controllers/BooksController.cs
using SelfAspNetApi.Models;
using System.Net;
using System.Web.Http;

namespace SelfAspNetApi.Controllers
{
    public class BooksController : ApiController
    {
        public Book GetBook(string id)
        {
            var db = new SelfAsp();
            var book = db.Book.Find(id);
            if (book == null)
            {
                throw new HttpResponseException(HttpStatusCode.NotFound);
            }
            return book;
        }
    }
}
